rest/data/project_settings_test.go

```diff
@@ -84,6 +84,52 @@ func TestSaveProjectSettingsForSectionForRepo(t *testing.T) {
 			assert.NotNil(t, newAdminFromDB)
 			assert.Contains(t, newAdminFromDB.Roles(), model.GetRepoAdminRole(ref.Id))
 		},
+		"Access with error": func(t *testing.T, ref model.RepoRef) {
```

Comment:
This test doe a lot of things that aren't relevant to what you need I think. This seems to test if markiing something restricted works, and it _also_ adds an admin. We should check the adding an admin case, removing an admin case, and the failure case. Right now I'm only seeing the success case but I'm not sure what error you're expecting at 99 (same with below)

Reply:
test made clearer